kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/commit/CommitPolicyConditions.scala

```diff
@@ -47,38 +50,32 @@ object Interval {
     val clock    = Clock.systemDefaultZone()
     Interval(interval, clock)
   }
+
+  private val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSSSSS")
 }
 
 case class Interval(interval: Duration, clock: Clock) extends CommitPolicyCondition {
 
-  override def eval(context: CommitContext, debugEnabled: Boolean): ConditionCommitResult = {
+  override def eval(context: CommitContext): ConditionCommitResult = {
     val nowInstant = clock.instant()
 
     val lastWriteInstant = Instant.ofEpochMilli(context.lastModified)
     val nextFlushTime    = lastWriteInstant.plus(interval)
-    val nextFlushDue     = nowInstant.isAfter(nextFlushTime) || nowInstant.equals(nextFlushTime)
+    val nextFlushDue     = nextFlushTime.toEpochMilli <= nowInstant.toEpochMilli
 
-    val logLine = Option.when(debugEnabled) {
-      val flushing      = if (nextFlushDue) "*" else ""
-      val timeRemaining = nextFlushTime.getEpochSecond - nowInstant.getEpochSecond
-      s"interval$flushing: {frequency:${interval.toSeconds}s, in:${timeRemaining}s, lastFlush:${lastWriteInstant.toString.substring(0,
```

Comment:
How would the user use the interval?

Reply:
1. Watching the log live.  See how long left until trigger kicks in so you know when to look for it in S3.
Does not require user to know the current time, just gives you the time left until it is triggered.
2. Analysing the logs later, eg using something like data dog.  I can see the rules which were executed.  It didn't fire yet, because it has x seconds left to go.  I can look forward in the log x seconds to see where it should be triggered.

Reply:
>Watching the log live. See how long left until trigger kicks in so you know when to look for it in S3.Does not require user to know the current time, just gives you the time left until it is triggered.

If the value says 243 seconds would the user start counting and wait 4mins+?

>Analysing the logs later, eg using something like data dog. I can see the rules which were executed. It didn't fire yet, because it has x seconds left to go. I can look forward in the log x seconds to see where it should be triggered.

DOesn't the timestamp of next flush give you the exact time of when it should happen next?

Reply:
> > Watching the log live. See how long left until trigger kicks in so you know when to look for it in S3.Does not require user to know the current time, just gives you the time left until it is triggered.
> 
> If the value says 243 seconds would the user start counting and wait 4mins+?

No.  But if it said 5 seconds, they might.  Why can't they both be there? This verbosity is designed to make the trigger events in the log easier to follow.  The easier these things are to understand for the customers, the less we will have to support this code.
 
> > Analysing the logs later, eg using something like data dog. I can see the rules which were executed. It didn't fire yet, because it has x seconds left to go. I can look forward in the log x seconds to see where it should be triggered.
> 
> DOesn't the timestamp of next flush give you the exact time of when it should happen next?

It does, that is useful too, but also I don't want to do the mental calculations of how much of my flush time has passed which is why I put it into the logs.